Lib/email/_header_value_parser.py

```diff
@@ -2626,7 +2626,7 @@ def _refold_parse_tree(parse_tree, *, policy):
                 want_encoding = False
                 last_ew = None
                 if part.syntactic_break:
-                    encoded_part = part.fold(policy=policy)[:-1] # strip nl
+                    encoded_part = part.fold(policy=policy)[:-len(policy.linesep)] # strip linesep
```

Comment:
If `policy.linesep` can be `None`:
```python
…[:-len(policy.linesep or "\n")]
```
If `policy.linesep` can be missing completely:
```python
…[:-len(policy.linesep if hasattr(policy, "linesep") else "\n")]
```
or a combination of both. I haven’t traced through the code completely to be sure of either scenario.